spec/features/user_adds_bar_spec.rb

```diff
@@ -0,0 +1,164 @@
+
+  feature "authenticated user adds a bar" do
+    scenario "successfully creates a new bar" do
+
+      new_user = User.create(email: "mckelvey.matt@gmail.com", password: "12345678")
+
+      visit user_session_path
+      fill_in 'Email', with: 'mckelvey.matt@gmail.com'
+      fill_in 'Password', with: '12345678'
+      click_button "Log in"
+      click_button "Add New Bar"
+      fill_in 'Name', with: 'Beantown Pub'
+      fill_in 'Description', with: 'After hours cocktail bar with pool.'
+      click_button "Add Bar"
+
+      expect(page).to have_content('Beantown Pub')
+    end
+
+    scenario "submits form without a name or description" do
+
+      new_user = User.create(email: "mckelvey.matt@gmail.com", password: "12345678")
+
+      visit user_session_path
+      fill_in 'Email', with: 'mckelvey.matt@gmail.com'
+      fill_in 'Password', with: '12345678'
+      click_button "Log in"
+      click_button "Add New Bar"
+      fill_in 'Name', with: ''
+      fill_in 'Description', with: ''
+      click_button "Add Bar"
+
+      expect(page).to have_content("Add Bar")
+      expect(page).to have_content("Name can't be blank. Description can't be blank")
+    end
+  end
+
+#   As an authenticated user
+#   I want to view a list of bars
+#   So that I can pick bars to review
+
+  feature "authenticated user views a list of bars" do
+    scenario "sees all bars" do
+
+      new_user = User.create(email: "mckelvey.matt@gmail.com", password: "12345678")
+      new_bar = Bar.create(name: "Beantown Pub", description: "Cool bar, guy.")
+
+      visit user_session_path
+      fill_in 'Email', with: 'mckelvey.matt@gmail.com'
+      fill_in 'Password', with: '12345678'
+      click_button "Log in"
+
+      expect(page).to have_content("All Bars")
+      expect(page).to have_content("Beantown Pub")
+    end
+  end
+#
+#   As an authenticated user
+#   I want to view the details about a bar
+#   So that I can get more information about the bar
+
+  feature "authenticated user views a bar's details" do
+    scenario "sees all details" do
+
+      new_user = User.create(email: "mckelvey.matt@gmail.com", password: "12345678")
+
+      visit user_session_path
+      fill_in 'Email', with: 'mckelvey.matt@gmail.com'
+      fill_in 'Password', with: '12345678'
+      click_button "Log in"
+      click_button "Add New Bar"
+      fill_in 'Name', with: 'Beantown Pub'
+      fill_in 'Description', with: 'After hours cocktail bar with pool.'
+      click_button "Add Bar"
+
+      expect(page).to have_content("Beantown Pub")
+      expect(page).to have_content("After hours cocktail bar with pool.")
+    end
+  end
+
+#   As an authenticated user
+#   I want to update a bar's information that I created
+#   So that I can correct errors or provide new information about the bar
+
+  feature "authenticated user edits an existing bar" do
+    scenario "successfully edits a bar" do
+
+      new_user = User.create(email: "mckelvey.matt@gmail.com", password: "12345678")
```

Comment:
I'd suggest looking into `FactoryGirl` to clean this up a bit.

Also, if you find yourself calling on `new_user` and `bar_1`, maybe try putting this outside the scenario:

```
let(:new_user) { User.create(....) }
let(:bar_1) { Bar.create(...) }
```
Then you can access the `new_user` and `bar_1` from all your scenarios